python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_categorical.py

```diff
@@ -0,0 +1,375 @@
+
+# %%
+# In order to deal with mixed continuous / discrete problems we can rely on the
+# ProductCovarianceModel class. We start here by defining the product kernel,
+# which combines SquaredExponential kernels for the continuous variables, and
+# LatentVariableModel for the discrete ones.
+
+# %%
+latdim = 1  # Dimension of the latent space
+activecoord = 1 + latdim * (lvls - 2)  # Nb ative coordinates in the latent space
+kx = ot.SquaredExponential(1)
+kz = otexp.LatentVariableModel(lvls, latdim)
+kLV = ot.ProductCovarianceModel([kx, kz])
+kLV.setNuggetFactor(1e-6)
+# Bounds for the hyperparameter optimization
+lb_LV = [1e-4] * dim + [-10] * activecoord
+ub_LV = [2.0] * dim + [10] * activecoord
+bounds_LV = ot.Interval(lb_LV, ub_LV)
+# Distribution for the hyperparameters initialization
+initdist_LV = ot.DistributionCollection()
+for i in range(len(lb_LV)):
+    initdist_LV.add(ot.Uniform(lb_LV[i], ub_LV[i]))
+initdist_LV = ot.ComposedDistribution(initdist_LV)
+
+# %%
+# As a reference, we consider a purely continuous kernel for independent Gaussian processes.
+# One for each combination of categorical variables levels.
+
+# %%
+k_ind = ot.SquaredExponential(1)
+lb_ind = [1e-4]
+ub_ind = [20]
+bounds_ind = ot.Interval(lb_ind, ub_ind)
+initdist_ind = ot.DistributionCollection()
+for i in range(len(lb_ind)):
+    initdist_ind.add(ot.Uniform(lb_ind[i], ub_ind[i]))
+initdist_ind = ot.ComposedDistribution(initdist_ind)
+initSample_ind = initdist_ind.getSample(10)
+optalg_ind = ot.MultiStart(ot.NLopt("LN_COBYLA"), initSample_ind)
+
+# %%
+# We generate the training data set
+X = dist.getSample(10)
+Y = fun(X)
+
+# And the plotting data set
+Xplt = dist.getSample(200)
+Xplt = Xplt.sort()
+Yplt = fun(Xplt)
+
+# %%
+# Initialize  and parameterize the optimization algorithm
+initSample_LV = initdist_LV.getSample(30)
+optalg_LV = ot.MultiStart(ot.NLopt("LN_COBYLA"), initSample_LV)
+
+# %%
+# Create and train the Gaussian process models
+basis = ot.ConstantBasisFactory(2).build()
+algo_LV = ot.KrigingAlgorithm(X, Y, kLV, basis)
+algo_LV.setOptimizationAlgorithm(optalg_LV)
+algo_LV.setOptimizationBounds(bounds_LV)
+algo_LV.run()
+res_LV = algo_LV.getResult()
+
+algo_ind_list = []
+for z in range(2):
+    # We select the training samples corresponding to the correct combination
+    # of categorical levels
+    ind = np.where(np.all(np.array(X[:, 1]) == z, axis=1))[0]
+    Xl = X[ind][:, 0]
+    Yl = Y[ind]
+
+    # Create and train the Gaussian process models
+    basis = ot.ConstantBasisFactory(1).build()
+    algo_ind = ot.KrigingAlgorithm(Xl, Yl, k_ind, basis)
+    algo_ind.setOptimizationAlgorithm(optalg_ind)
+    algo_ind.setOptimizationBounds(bounds_ind)
+    algo_ind.run()
+    algo_ind_list.append(algo_ind.getResult())
+
+# %%
+# We plot the prediction of the mixed continuous / categorical GP,
+# as well as the one of the two separate continuous GPs
+fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True, figsize=(15, 10))
```

Comment:
This should be done using [GridLayout](http://openturns.github.io/openturns/latest/user_manual/_generated/openturns.GridLayout.html).

Reply:
As far as I understand, you do not want to use the graphics OpenTURNS API within Python examples. What not to discuss that topic at the next technical meeting?